db/crud.go

```diff
@@ -672,6 +672,7 @@ func (db *Database) PutExistingRev(docid string, body Body, docHistory []string)
 		}
 		if currentRevIndex == 0 {
 			base.LogTo("CRUD+", "PutExistingRev(%q): No new revisions to add", docid)
+			body["_rev"] = newRev
```

Comment:
The scoping is very hard to follow here.  Below when we update body["_rev"] we're about to pass body back to updateDoc via the callback.  Here we're directly updating the body for use in the PutExistingRev scope.  I can see how it might work for this issue, but it's difficult to understand the intention (and why we don't set "_rev" in other cases).

I think it's ok to leave for now, though - maybe just with an additional comment for clarification.  We're planning to refactor out most of these usages of _rev and _id in the document body in 2.0 anyway, so we can revisit at that point.

Reply:
Adding a comment